File: java/Finish.java
import android.arch.lifecycle.LifecycleOwner;

public class Finish extends SingleLiveEvent<Class> {

    public void observe(LifecycleOwner owner, final FinishObserver observer) {
        super.observe(owner, s -> {
            if (s == null) {
                return;
            }

            observer.onFinish(s);
        });
    }

    public interface FinishObserver {

        void onFinish(Class clazz);
    }
}

File: kotlin/Finish.kt
package dev.dita.maziwapp.ui.base.v2

import android.arch.lifecycle.LifecycleOwner
import android.arch.lifecycle.Observer

class Finish : SingleLiveEvent<Class<Any>>() {
    open fun observe(owner: LifecycleOwner, observer: FinishObserver) {
        super.observe(owner, Observer {
            if (it == null) {
                return@Observer
            }

            observer.onFinish(it)
        })
    }

    open interface FinishObserver {
        fun onFinish(clazz: Class<Any>)
    }
}